### ISSUE TITLE
Segmentation fault in testsuite

### ISSUE THREAD

**Issue by ghisvail** [clMathLibraries/clFFT#161]
I tried running the testsuite on a laptop running up-to-date Debian testing, using the `2.12.1` tagged version as well as the latest tip of `develop`. I had to apply a patch to get around #160 first.

OpenCL info:
```
clFFT client API version:  2.12.1
clFFT runtime version:     2.12.1

    CL_PLATFORM_PROFILE:     FULL_PROFILE
    CL_PLATFORM_VERSION:     OpenCL 2.0 pocl 0.13, LLVM 3.8.1
    CL_PLATFORM_NAME:        Portable Computing Language
    CL_PLATFORM_VENDOR:      The pocl project
    CL_PLATFORM_EXTENSIONS:  cl_khr_icd

    CL_DEVICE_NAME:                      pthread-Intel(R) Core(TM) i5-4300M CPU @ 2.60GHz
    CL_DEVICE_VERSION:                   OpenCL 2.0 pocl
    CL_DRIVER_VERSION:                   0.13
    CL_DEVICE_TYPE:                      defaultCPU
    CL_DEVICE_MAX_CLOCK_FREQUENCY:       3300
    CL_DEVICE_ADDRESS_BITS:              64
    CL_DEVICE_AVAILABLE:                 TRUE
    CL_DEVICE_COMPILER_AVAILABLE:        TRUE
    CL_DEVICE_OPENCL_C_VERSION:          OpenCL C 2.0
    CL_DEVICE_MAX_WORK_GROUP_SIZE:       4096
    CL_DEVICE_MAX_WORK_ITEM_DIMENSIONS:  3
                         Dimension[ 0 ]  4096
                         Dimension[ 1 ]  4096
                         Dimension[ 2 ]  4096
    CL_DEVICE_HOST_UNIFIED_MEMORY:       TRUE
    CL_DEVICE_MAX_CONSTANT_BUFFER_SIZE:  10424524800 ( 10180200 KB )
    CL_DEVICE_LOCAL_MEM_SIZE:            10424524800 ( 10180200 KB )
    CL_DEVICE_GLOBAL_MEM_SIZE:           10424524800 ( 9941 MB )
    CL_DEVICE_MAX_MEM_ALLOC_SIZE:        10424524800 ( 9941 MB )
    CL_DEVICE_EXTENSIONS:                cl_khr_byte_addressable_store cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_spir cl_khr_int64 cl_khr_fp64 cl_khr_int64_base_atomics cl_khr_int64_extended_atomics
```

output of `./Test --short`:
```
[ RUN      ] accuracy_test_pow2_double.large_2D_in_place_real_to_hermitian_interleaved
Segmentation fault
```

applied patch:
```
From 3e5e5c5da7c87c1837128b82a64f61f8978cc750 Mon Sep 17 00:00:00 2001
From: Ghislain Antony Vaillant <ghisvail@gmail.com>
Date: Thu, 4 Aug 2016 10:19:00 +0100
Subject: [PATCH] Bypass execution of memory intensive testcases

---
 src/tests/accuracy_test_postcallback.cpp     | 40 ++++++++++++++--------------
 src/tests/accuracy_test_pow2.cpp             | 22 +++++++--------
 src/tests/accuracy_test_pow2_precallback.cpp | 22 +++++++--------
 3 files changed, 42 insertions(+), 42 deletions(-)

diff --git a/src/tests/accuracy_test_postcallback.cpp b/src/tests/accuracy_test_postcallback.cpp
index ef03a19..58a2bb3 100644
--- a/src/tests/accuracy_test_postcallback.cpp
+++ b/src/tests/accuracy_test_postcallback.cpp
@@ -1485,17 +1485,17 @@ void pow2_large_1D_4M_in_place_real_to_hermitian_interleaved()
 	postcallback_real_to_complex<T, cl_T, fftw_T>( pattern, lengths, batch, input_strides, output_strides, input_distance, output_distance, layout, placeness );
 }
 
-TEST_F(accuracy_test_postcallback_single, pow2_large_1D_4M_in_place_real_to_hermitian_interleaved)
-{
-	try { pow2_large_1D_4M_in_place_real_to_hermitian_interleaved< float, cl_float, fftwf_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
+//TEST_F(accuracy_test_postcallback_single, pow2_large_1D_4M_in_place_real_to_hermitian_interleaved)
+//{
+	//try { pow2_large_1D_4M_in_place_real_to_hermitian_interleaved< float, cl_float, fftwf_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
 
-TEST_F(accuracy_test_postcallback_double, pow2_large_1D_4M_in_place_real_to_hermitian_interleaved)
-{
-	try { pow2_large_1D_4M_in_place_real_to_hermitian_interleaved< double, cl_double, fftw_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
+//TEST_F(accuracy_test_postcallback_double, pow2_large_1D_4M_in_place_real_to_hermitian_interleaved)
+//{
+	//try { pow2_large_1D_4M_in_place_real_to_hermitian_interleaved< double, cl_double, fftw_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
 
 template< class T, class cl_T, class fftw_T >
 void pow2_large_1D_4M_out_of_place_real_to_hermitian_planar()
@@ -1514,17 +1514,17 @@ void pow2_large_1D_4M_out_of_place_real_to_hermitian_planar()
 	postcallback_real_to_complex<T, cl_T, fftw_T>( pattern, lengths, batch, input_strides, output_strides, input_distance, output_distance, layout, placeness );
 }
 
-TEST_F(accuracy_test_postcallback_single, pow2_large_1D_4M_out_of_place_real_to_hermitian_planar)
-{
-	try { pow2_large_1D_4M_out_of_place_real_to_hermitian_planar< float, cl_float, fftwf_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
+//TEST_F(accuracy_test_postcallback_single, pow2_large_1D_4M_out_of_place_real_to_hermitian_planar)
+//{
+	//try { pow2_large_1D_4M_out_of_place_real_to_hermitian_planar< float, cl_float, fftwf_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
 
-TEST_F(accuracy_test_postcallback_double, pow2_large_1D_4M_out_of_place_real_to_hermitian_planar)
-{
-	try { pow2_large_1D_4M_out_of_place_real_to_hermitian_planar< double, cl_double, fftw_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
+//TEST_F(accuracy_test_postcallback_double, pow2_large_1D_4M_out_of_place_real_to_hermitian_planar)
+//{
+	//try { pow2_large_1D_4M_out_of_place_real_to_hermitian_planar< double, cl_double, fftw_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
 
 template< class T, class cl_T, class fftw_T >
 void pow3_large_1D_out_of_place_real_to_hermitian_planar()
diff --git a/src/tests/accuracy_test_pow2.cpp b/src/tests/accuracy_test_pow2.cpp
index c3b0a84..a81e02f 100644
--- a/src/tests/accuracy_test_pow2.cpp
+++ b/src/tests/accuracy_test_pow2.cpp
@@ -1793,17 +1793,17 @@ void large_1D_4M_in_place_real_to_hermitian_interleaved()
 	real_to_complex<T, cl_T, fftw_T>( pattern, lengths, batch, input_strides, output_strides, input_distance, output_distance, layout, placeness );
 }
 
-TEST_F(accuracy_test_pow2_single, large_1D_4M_in_place_real_to_hermitian_interleaved)
-{
-	try { large_1D_4M_in_place_real_to_hermitian_interleaved< float, cl_float, fftwf_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
-
-TEST_F(accuracy_test_pow2_double, large_1D_4M_in_place_real_to_hermitian_interleaved)
-{
-	try { large_1D_4M_in_place_real_to_hermitian_interleaved< double, cl_double, fftw_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
+//TEST_F(accuracy_test_pow2_single, large_1D_4M_in_place_real_to_hermitian_interleaved)
+//{
+	//try { large_1D_4M_in_place_real_to_hermitian_interleaved< float, cl_float, fftwf_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
+
+//TEST_F(accuracy_test_pow2_double, large_1D_4M_in_place_real_to_hermitian_interleaved)
+//{
+	//try { large_1D_4M_in_place_real_to_hermitian_interleaved< double, cl_double, fftw_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
 // *****************************************************
 // *****************************************************
 template< class T, class cl_T, class fftw_T >
diff --git a/src/tests/accuracy_test_pow2_precallback.cpp b/src/tests/accuracy_test_pow2_precallback.cpp
index 9bb78bf..4e519a0 100644
--- a/src/tests/accuracy_test_pow2_precallback.cpp
+++ b/src/tests/accuracy_test_pow2_precallback.cpp
@@ -1851,17 +1851,17 @@ void large_1D_4M_in_place_real_to_hermitian_interleaved()
 	precallback_real_to_complex<T, cl_T, fftw_T>( pattern, lengths, batch, input_strides, output_strides, input_distance, output_distance, layout, placeness );
 }
 
-TEST_F(accuracy_test_pow2_precallback_single, large_1D_4M_in_place_real_to_hermitian_interleaved)
-{
-	try { large_1D_4M_in_place_real_to_hermitian_interleaved< float, cl_float, fftwf_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
-
-TEST_F(accuracy_test_pow2_precallback_double, large_1D_4M_in_place_real_to_hermitian_interleaved)
-{
-	try { large_1D_4M_in_place_real_to_hermitian_interleaved< double, cl_double, fftw_complex >(); }
-	catch( const std::exception& err ) { handle_exception(err);	}
-}
+//TEST_F(accuracy_test_pow2_precallback_single, large_1D_4M_in_place_real_to_hermitian_interleaved)
+//{
+	//try { large_1D_4M_in_place_real_to_hermitian_interleaved< float, cl_float, fftwf_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
+
+//TEST_F(accuracy_test_pow2_precallback_double, large_1D_4M_in_place_real_to_hermitian_interleaved)
+//{
+	//try { large_1D_4M_in_place_real_to_hermitian_interleaved< double, cl_double, fftw_complex >(); }
+	//catch( const std::exception& err ) { handle_exception(err);	}
+//}
 // *****************************************************
 // *****************************************************
 template< class T, class cl_T, class fftw_T >
-- 
2.8.1
```

**Comment by bragadeesh** [comment]
The 'CL_DEVICE_LOCAL_MEM_SIZE' shown for POCL is huge. It is equal to the total global memory size available. I am suspecting this may be causing issues such as memory leaks & seg faults.

You would have to step through clFFT source in order to investigate and fix this. If you can change code in plan.cpp in the function FFTPlan::SetEnvelope () like the following and rebuild clFFT and test, it would be great. 

Near the end of the function `FFTPlan::SetEnvelope ()` there is a line `BUG_CHECK (envelope.limit_LocalMemSize >= 1024)`  .Please add the following code near that.

```c
envelope.limit_LocalMemSize = 32768;
envelope.limit_Dimensions = 3;
envelope.limit_WorkGroupSize = 256;
envelope.limit_Size[0] = 256;
envelope.limit_Size[1] = 256;
envelope.limit_Size[2] = 256;
BUG_CHECK (envelope.limit_LocalMemSize >= 1024);
```

If this change fixes things, then it would tell us how to change clFFT to work with POCL.

**Comment by ghisvail** [comment]
clinfo for `GeForce GT 730M`:
```
  Platform Name                                   NVIDIA CUDA
Number of devices                                 1
  Device Name                                     GeForce GT 730M
  Device Vendor                                   NVIDIA Corporation
  Device Vendor ID                                0x10de
  Device Version                                  OpenCL 1.2 CUDA
  Driver Version                                  352.79
  Device OpenCL C Version                         OpenCL C 1.2 
  Device Type                                     GPU
  Device Profile                                  FULL_PROFILE
  Device Topology (NV)                            PCI-E, 02:00.0
  Max compute units                               2
  Max clock frequency                             758MHz
  Compute Capability (NV)                         3.5
  Device Partition                                (core)
    Max number of sub-devices                     1
    Supported partition types                     None
  Max work item dimensions                        3
  Max work item sizes                             1024x1024x64
  Max work group size                             1024
  Preferred work group size multiple              32
  Warp size (NV)                                  32
  Preferred / native vector sizes                 
    char                                                 1 / 1       
    short                                                1 / 1       
    int                                                  1 / 1       
    long                                                 1 / 1       
    half                                                 0 / 0        (n/a)
    float                                                1 / 1       
    double                                               1 / 1        (cl_khr_fp64)
  Half-precision Floating-point support           (n/a)
  Single-precision Floating-point support         (core)
    Denormals                                     Yes
    Infinity and NANs                             Yes
    Round to nearest                              Yes
    Round to zero                                 Yes
    Round to infinity                             Yes
    IEEE754-2008 fused multiply-add               Yes
    Support is emulated in software               No
    Correctly-rounded divide and sqrt operations  Yes
  Double-precision Floating-point support         (cl_khr_fp64)
    Denormals                                     Yes
    Infinity and NANs                             Yes
    Round to nearest                              Yes
    Round to zero                                 Yes
    Round to infinity                             Yes
    IEEE754-2008 fused multiply-add               Yes
    Support is emulated in software               No
    Correctly-rounded divide and sqrt operations  No
  Address bits                                    64, Little-Endian
  Global memory size                              1073610752 (1024MiB)
  Error Correction support                        No
  Max memory allocation                           268402688 (256MiB)
  Unified memory for Host and Device              No
  Integrated memory (NV)                          No
  Minimum alignment for any data type             128 bytes
  Alignment of base address                       4096 bits (512 bytes)
  Global Memory cache type                        Read/Write
  Global Memory cache size                        32768
  Global Memory cache line                        128 bytes
  Image support                                   Yes
    Max number of samplers per kernel             32
    Max size for 1D images from buffer            134217728 pixels
    Max 1D or 2D image array size                 2048 images
    Max 2D image size                             16384x16384 pixels
    Max 3D image size                             4096x4096x4096 pixels
    Max number of read image args                 256
    Max number of write image args                16
  Local memory type                               Local
  Local memory size                               49152 (48KiB)
  Registers per block (NV)                        65536
  Max constant buffer size                        65536 (64KiB)
  Max number of constant args                     9
  Max size of kernel argument                     4352 (4.25KiB)
  Queue properties                                
    Out-of-order execution                        Yes
    Profiling                                     Yes
  Prefer user sync for interop                    No
  Profiling timer resolution                      1000ns
  Execution capabilities                          
    Run OpenCL kernels                            Yes
    Run native kernels                            No
    Kernel execution timeout (NV)                 No
  Concurrent copy and kernel execution (NV)       Yes
    Number of async copy engines                  1
  printf() buffer size                            1048576 (1024KiB)
  Built-in kernels                                
  Device Available                                Yes
  Compiler Available                              Yes
  Linker Available                                Yes
  Device Extensions                               cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_copy_opts  cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64 
```

**Comment by ghisvail** [comment]
clinfo for `Tesla K40c`:
```
  Platform Name:				 NVIDIA CUDA
Number of devices:				 2
  Device Type:					 CL_DEVICE_TYPE_GPU
  Device ID:					 4318
  Max compute units:				 15
  Max work items dimensions:			 3
    Max work items[0]:				 1024
    Max work items[1]:				 1024
    Max work items[2]:				 64
  Max work group size:				 1024
  Preferred vector width char:			 1
  Preferred vector width short:			 1
  Preferred vector width int:			 1
  Preferred vector width long:			 1
  Preferred vector width float:			 1
  Preferred vector width double:		 1
  Native vector width char:			 1
  Native vector width short:			 1
  Native vector width int:			 1
  Native vector width long:			 1
  Native vector width float:			 1
  Native vector width double:			 1
  Max clock frequency:				 745Mhz
  Address bits:					 64
  Max memory allocation:			 3019784192
  Image support:				 Yes
  Max number of images read arguments:		 256
  Max number of images write arguments:		 16
  Max image 2D width:				 16384
  Max image 2D height:				 16384
  Max image 3D width:				 4096
  Max image 3D height:				 4096
  Max image 3D depth:				 4096
  Max samplers within kernel:			 32
  Max size of kernel argument:			 4352
  Alignment (bits) of base address:		 4096
  Minimum alignment (bytes) for any datatype:	 128
  Single precision floating point capability
    Denorms:					 Yes
    Quiet NaNs:					 Yes
    Round to nearest even:			 Yes
    Round to zero:				 Yes
    Round to +ve and infinity:			 Yes
    IEEE754-2008 fused multiply-add:		 Yes
  Cache type:					 Read/Write
  Cache line size:				 128
  Cache size:					 245760
  Global memory size:				 12079136768
  Constant buffer size:				 65536
  Max number of constant args:			 9
  Local memory type:				 Local
  Local memory size:				 49152
  Error correction support:			 1
  Unified memory for Host and Device:		 0
  Profiling timer resolution:			 1000
  Device endianess:				 Little
  Available:					 Yes
  Compiler available:				 Yes
  Execution capabilities:				 
    Execute OpenCL kernels:			 Yes
    Execute native function:			 No
  Queue properties:				 
    Out-of-Order:				 Yes
    Profiling :					 Yes
  Platform ID:					 0x2551010
  Name:						 Tesla K40c
  Vendor:					 NVIDIA Corporation
  Device OpenCL C version:			 OpenCL C 1.2 
  Driver version:				 352.63
  Profile:					 FULL_PROFILE
  Version:					 OpenCL 1.2 CUDA
  Extensions:					 cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_copy_opts  cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64
```

**Comment by ghisvail** [comment]
After applying the suggested change on the latest tip of `develop`, the testsuite ran without segfaults nor memory issues. Some tests did fail though. Here is the summary:
```
[----------] Global test environment tear-down
[==========] 882 tests from 4 test cases ran. (1301163 ms total)
[  PASSED  ] 862 tests.
[  FAILED  ] 20 tests, listed below:
[  FAILED  ] accuracy_test_pow2_single.large_1D_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_single.large_1D_4M_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_single.large_1D_out_of_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_single.large_1D_out_of_place_real_to_hermitian_planar
[  FAILED  ] accuracy_test_pow2_single.large_1D_round_trip_real_to_complex
[  FAILED  ] accuracy_test_pow2_double.large_1D_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_double.large_1D_4M_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_double.large_1D_out_of_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_double.large_1D_out_of_place_real_to_hermitian_planar
[  FAILED  ] accuracy_test_pow2_double.large_1D_round_trip_real_to_complex
[  FAILED  ] accuracy_test_pow2_precallback_single.large_1D_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_precallback_single.large_1D_4M_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_precallback_single.large_1D_out_of_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_precallback_single.large_1D_out_of_place_real_to_hermitian_planar
[  FAILED  ] accuracy_test_pow2_precallback_single.large_1D_round_trip_real_to_complex
[  FAILED  ] accuracy_test_pow2_precallback_double.large_1D_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_precallback_double.large_1D_4M_in_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_precallback_double.large_1D_out_of_place_real_to_hermitian_interleaved
[  FAILED  ] accuracy_test_pow2_precallback_double.large_1D_out_of_place_real_to_hermitian_planar
[  FAILED  ] accuracy_test_pow2_precallback_double.large_1D_round_trip_real_to_complex

20 FAILED TESTS
  YOU HAVE 32 DISABLED TESTS
```

**Comment by bragadeesh** [comment]
Ok, thats great, that confirms what I was suspecting. I will fix the code in the library appropriately.

For the seg-fault on NV & failure with pocl, it appears that all the problem are in large 1d real FFTs tests. A seg-fault likely happens due to kernels writing to out-of-bounds memory location. The NV cl info does not give any clues to me yet. This would have to be investigated at the kernel level on NV hardware. Unfortunately, I don't have the time to do that at this point.

But to get past the issue for now, we can try to disable a particular real FFT optimization and see if the tests would functionally pass. On the tip of develop branch (commit id 0f658cc), in plan.cpp file at line 1229, there is an if condition. We need to make it evaluate to false. If you can change it like this:

```c
if( 0 && (fftPlan->inputLayout == CLFFT_REAL) && IsPo2(fftPlan->length[0])
	&& (fftPlan->length.size() == 1)
	&& (fftPlan->inStride[0] == 1) && (fftPlan->outStride[0] == 1)
	&& (fftPlan->length[0] > 4096) && (fftPlan->length.size() == 1) )
```

please try that change and see if it fixes the seg-fault.

**Comment by ghisvail** [comment]
For the `Nvidia GT-730M`, I still get the rise of memory consumption until I get a "Data too large" exception in the `accuracy_test_pow2_precallback` testcases.
```
[ RUN      ] accuracy_test_pow2_precallback_single.normal_2D_forward_in_place_complex_planar_to_complex_planar
[       OK ] accuracy_test_pow2_precallback_single.normal_2D_forward_in_place_complex_planar_to_complex_planar (398 ms)
[ RUN      ] accuracy_test_pow2_precallback_single.normal_2D_backward_in_place_complex_planar_to_complex_planar
OPENCL_V_THROWERROR< CLFFT_MEM_OBJECT_ALLOCATION_FAILURE > (1097): writing input buffer - planar imaginary ( ::clEnqueueWriteBuffer() )
--- Exception caught ---
Data set is too large for this device -- skipping test
[...]
```

**Comment by ghisvail** [comment]
For `pocl`, I don't get any segfaults anymore, just tests not passing.

**Comment by ghisvail** [comment]
I only tried the testsuite with the --short option however. I can try launching the full testsuite with `pocl` if it does not take too much time to execute?

**Comment by bragadeesh** [comment]
The full tests can ran for hours or even more than a day depending on hardware & how well optimizations are working on that HW+OpenCL runtime;

The code patch I suggested earlier is to see if it fixes both the fails on pocl & the seg-fault on NV.

**Comment by bragadeesh** [comment]
Closing this issue for now .. please update with findings

**Comment by ghisvail** [comment]
I don't understand the rationales behind closing an issue, which solution is currently pending. I have provided enough evidence to confirm the existence of the issue and triage it, but AFAIC a proper fix has yet to be found and merged.

> Closing this issue for now .. please update with findings

Please consider closing this issue only after a proper fix has been committed. Thanks.

**Comment by bragadeesh** [comment]
Did you try the workaround I suggested? any progress on the NV machine?

**Comment by ghisvail** [comment]
I'll find time for it.

**Comment by ghisvail** [comment]
I applied your suggested fixes on a dedicated branch https://github.com/ghisvail/clFFT/commit/cc6ab54f588bc152c3808b130be7fb7de9396149.

All radix-2 tests now pass on `pocl`, I have not tried on `nv` yet.

**Comment by ghisvail** [comment]
Still failing on the `nvidia` card. Please re-open.

**Comment by bragadeesh** [comment]
Can you specify what is failing on NV? 

I just compiled clFFT develop tip + real FFT opt-disable fix mentioned above in this thread, on GTX 980 and ran all pow2 tests. All of the regular (non callback tests) tests passed. Part of the callback tests also passed, the rest are not running due to context creation failure. But when I run those tests individually, they run to completion and pass correctly. I am attaching a log of the run. 
[log_linux_gtx980.txt](https://github.com/clMathLibraries/clFFT/files/440341/log_linux_gtx980.txt)


This appears like some problem with NV's opencl runtime.  You would have to contact Nvidia to look at this issue.

**Comment by ghisvail** [comment]
> Can you specify what is failing on NV?

Same context creation errors as your reported.

How come these fail when running the testsuite entrely but succeeds when run individually then?

> This appears like some problem with NV's opencl runtime. You would have to contact Nvidia to look at this issue.

Judging by the symptoms above, one might be more enclined to think this is an issue with the design of the testsuite. Before raising our forks on the `nvidia` people, more solid triaging may be required, don't you think?

And please **keep the issue open until fixed**, so that it is visible. Other people might care about this issue too, and leaving this closed does no good in getting more support involved. Thanks.

**Comment by bragadeesh** [comment]
The fact that the test-suite works fine on AMD, Intel runtimes and as you have tested POCL runtimes and works fine on AMD & Intel hardware, tells me it is highly unlikely for the test suite to have design issues. 
Anyway, we don't have much cycles to look into this; and I hope you/open-source community is able to triage this.

This particular issue needs to remain closed as the title is generic; if you would like you can open a new issue adding "issues on Nvidia platform" in the title. And we can reference the discussion here in that.

**Comment by bragadeesh** [comment]
I tested with only Intel's drivers; not beignet. Sure, you can open another issue for that ..
